controllers/argocd/statefulset.go

```diff
@@ -441,13 +442,61 @@ func getArgoControllerContainerEnv(cr *argoprojv1a1.ArgoCD) []corev1.EnvVar {
 	return env
 }
 
-func (r *ReconcileArgoCD) reconcileApplicationControllerStatefulSet(cr *argoprojv1a1.ArgoCD, useTLSForRedis bool) error {
+func (r *ReconcileArgoCD) getArgoControllerReplicaCount(cr *argoprojv1a1.ArgoCD) int32 {
 	var replicas int32 = common.ArgocdApplicationControllerDefaultReplicas
+	var minShards int32 = cr.Spec.Controller.Sharding.MinShards
+	var maxShards int32 = cr.Spec.Controller.Sharding.MaxShards
+
+	if cr.Spec.Controller.Sharding.DynamicScalingEnabled {
+		if minShards < 1 {
+			err := errors.New("incorrect configuration")
+			log.Error(err, "Minimum number of shards cannot be less than 1. Setting default value to 1")
+			minShards = 1
+		}
+
+		if maxShards < minShards {
+			err := errors.New("incorrect configuration")
+			log.Error(err, "Maximum number of shards cannot be less than minimum number of shards. Setting maximum shards same as minimum shards")
+			maxShards = minShards
+		}
+
+		clustersPerShard := cr.Spec.Controller.Sharding.ClustersPerShard
+		if clustersPerShard < 1 {
```

Comment:
Is this check really needed ? As the `ValidatingWebHook` should restrict such invalid configuration.